lib/evmone/eof.cpp

```diff
@@ -90,37 +150,270 @@ std::variant<EOFSectionHeaders, EOFValidationError> validate_eof_headers(
     if (state != State::terminated)
         return EOFValidationError::section_headers_not_terminated;
 
-    const auto section_bodies_size = section_headers[CODE_SECTION] + section_headers[DATA_SECTION];
+    const auto section_bodies_size =
+        (!section_headers[TYPE_SECTION].empty() ? section_headers[TYPE_SECTION].front() : 0) +
```

Comment:
Type section is mandatory
```suggestion
        section_headers[TYPE_SECTION].front() +
```